### ISSUE TITLE
"Error: Unexpected end of document" when blocking trackers

### ISSUE THREAD

**Issue by Bazoogle** [Hamza417/Inure#269]
**Describe the bug**
When blocking trackers, either by going to _Apps -> Selecting an App -> Trackers -> Selecting the "batch select" button in the top left -> Selecting block_
**OR**
Using the new batch tracker block feature

Sometimes I will get an error popup at the bottom that says "Error: Unexpected end of document". When I used the batch tracker block for the first time, the message actually popped up like 20 times layered one over the other. But when I did them manually one by one, it would only happen once. It's inconsistent, and not easy to replicate. After the 20 messages popped up the first time with the batch block, I did the block a second time, and it did not pop up that time.
**To Reproduce**
1. Go to Apps
2. Select an App
3. Click Trackers
4. Select the "batch selection" icon in the top left
5. Select block
6. Error pops up in the bottom sometimes

OR

1. Go to Batch
2. Select all apps
3. Click the nuclear "tracker" icon
4. Select all trackers
5. Block
6. Error pops up in the bottom sometimes, potentially several times

Reproducing is, unfortunately, inconsistent. I think I sometimes noticed it with the play store app "AppSales" and a few others. It seems after the first time, it doesn't happen as often. I was able to get it to happen again by re-enable the trackers, clearing the app data for Inure, then trying again. Though I then tried to do that a second time, and it didn't happen that time.

**Expected behavior**
Error should not pop up, and all selected trackers should be blocked

<details>

   <summary><b>Screenshots</b></summary>

   ![image](https://github.com/Hamza417/Inure/assets/8096843/9ce7d25b-5786-4252-9cff-eda7c84d15da)
</details>

**Smartphone:**
 - Device: Moto Edge 2021
 - OS: LineageOS 20 Android 13
 - Version: Build 65

**Comment by Hamza417** [comment]
Do you use any other program to block app components?

**Comment by Hamza417** [comment]
> Do you use any other program to block app components?

I ran the blocker several times and haven't been able to reproduce this problem, could you answer this?

**Comment by Bazoogle** [comment]
> Do you use any other program to block app components?

I am not using any other app to block trackers within an app. I previously used Warden, though I just did a clean install and didn't install Warden and only used Inure. I use AdAway to block it via hosts, so if that could interfere, then yes. If that's separate, then I don't use any other service.

I could try and recreate it once again and get a log. I tried enabling debugging and viewing logcat. Is there a way I can get a verbose log?
